receiver/snmpreceiver/scraper.go

```diff
@@ -16,19 +16,70 @@ package snmpreceiver // import "github.com/open-telemetry/opentelemetry-collecto
 
 import (
 	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/receiver/scrapererror"
 	"go.uber.org/zap"
 )
 
+const generalResourceKey = ""
+
+var (
+	// Error messages
+	errMsgBadValueType                   = `returned metric SNMP data type for OID '%s' is not supported`
+	errMsgIndexedBadValueType            = `returned metric SNMP data type for OID '%s' from column OID '%s' is not supported`
+	errMsgIndexedAttributesBadValueType  = `returned attribute SNMP data type for OID '%s' from column OID '%s' is not supported`
+	errMsgOIDAttributeEmptyValue         = `not creating indexed metric '%s' datapoint as metric OID attribute value is blank`
+	errMsgOIDResourceAttributeEmptyValue = `not creating indexed metric '%s' or resource as related resource attribute value is blank`
+)
+
 // snmpScraper handles scraping of SNMP metrics
 type snmpScraper struct {
+	client   client
 	logger   *zap.Logger
 	cfg      *Config
 	settings component.ReceiverCreateSettings
 }
 
+// metricsByResourceContainer is a helper struct to hold metric data keyed by resource key and metric name
+type metricsByResourceContainer struct {
```

Comment:
@djaglowski i get that this is a little bulky for what it is, but i added it as direct feedback that using nested maps was a little confusing. making an alias instead would help too (not having to see function arguments as map[string]map[string]something, but the usage "might" be a little less clear to look at. i'm not opposed to changing if you think it would be better (like i said, i recognize that this isn't very pretty for what it is), just wanted to throw that out there first.

Reply:
@djaglowski took me a bit to understand what you were suggesting, but after playing around a bit i figured it out 👍